mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -256,6 +256,15 @@ class LogisticRegression @Since("1.2.0") (
   @Since("1.5.0")
   override def getThresholds: Array[Double] = super.getThresholds
 
+  /**
+   * Set suggested depth for treeAggregate or treeReduce (>= 2).
+   * Default is 2.
+   * @group setParam
```

Comment:
I'm not suggesting you implement the logic to automatically decide what depth is best. Typically parameters like these that require a user to understand how this algorithm is implemented (i.e. that it uses a treeAggregation to collect gradient updates) we make "expertParam."